### ISSUE TITLE
Automatic retry if login fails

### ISSUE THREAD

**Issue by luc-ass** [luc-ass/homebridge-evohome#136]
As this plugin is now able to be run as a child bridge (and thus not loading until it is logged in), an automatic retry to log in should be implemented.

#### Problematic behaviour 
Login fails → Plugin fails

#### Expected behaviour
Login fails → Plugin reports error → some timeout function with safe delay → retry

#### Additional information
This error is tricky to reproduce, as hitting the rate limiter on honeywells servers is not very reliable.

#### TODO
- [ ] determine a way to retry the login
- [ ] determine a safe timeout, so we don't keep hitting the rate limiter

**Comment by luc-ass** [comment]
Implemented in the rewrite, released in `1.0.0-beta.1`.

- A failed login is retried with exponential backoff and jitter instead of
  killing the plugin until a manual restart.
- An expired token is refreshed, and if the refresh fails the plugin logs in
  again rather than staying dead — that was the actual failure mode behind this
  ticket.
- Tokens are cached on disk between restarts, so a restart does not cost a fresh
  login and therefore does not walk into the rate limiter.
- The backoff was chosen so that a rate-limited account backs off instead of
  hammering the endpoint: Honeywell's limit applies per account, so a badly
  behaved plugin degrades every other client on the same login.

Accessories now survive a failed start as well: they stay registered and show as
unreachable, instead of the `callback([])` in 0.11.2 that made HomeKit forget
them.

**Comment by PuzzledUser** [comment]
Aha! That explains why I've been seeing this in the logs every half hour for weeks! 🤣
```
...
[08/09/2026, 16:39:03] [Evohome] Renewing Honeywell API authentication token failed: HTTP error 400
[08/09/2026, 17:08:32] [Evohome] Renewing Honeywell API authentication token failed: HTTP error 400
[08/09/2026, 17:38:01] [Evohome] Renewing Honeywell API authentication token failed: HTTP error 400
```
A Homebridge restart fixed it whilst I gird my loins to move to V2(/v1). Just one glitch overnight:
```
[09/09/2026, 01:09:40] [Evohome] Renewing Honeywell API authentication token failed: HTTP error 504
```

**Comment by luc-ass** [comment]
@PuzzledUser That log is this ticket, exactly as it was written.

In 0.11.2 `renewSession()` only logged the failure. The spent refresh token
stayed in the session object, the renewal ran on a fixed interval derived from
the first `expires_in`, and so the same dead token was presented every half hour
for as long as Homebridge ran — while every API call kept using the expired
access token. A restart was the only way out, which is why one fixed it.

1.0.0 refreshes on demand rather than on a timer, and a refused refresh falls
through to a full login instead of being logged and forgotten.

**Your log made me re-read that fallback, and it was not complete.** 1.0.0 took
the login route only when the response body carried `invalid_grant`. Yours shows
no code at all, just `HTTP error 400` — and with that body 1.0.0 would have
treated the refusal as temporary and gone on presenting the same refresh token,
behind a backoff, but with no way out either. Fixed now: any answer that judges
the token — any 4xx, and the HTTP 200 with an `error` field the TCC API also
uses — leads to a login. Only 429 and 5xx are repeated, because those say
nothing about the token, and spending a password login on them would just add
pressure to Honeywell's per-account rate limit.

The 504 overnight is unrelated and needs nothing: that is Honeywell's server
(same as #185). 1.0.0 counts 5xx as temporary and retries with backoff.

The fix is in #229.

**Comment by PuzzledUser** [comment]
Yes, I understood the aim and that you'd addressed the logging. I shared this as I was excited that I now knew why the 400s were repeating. 🎉

Good you're continuing to refine. My version also just shows the full (400) failure reason the first time (weeks ago now 🤣), but then tones it down until after a success.

I'll join the V1 party at some point... 
Thanks
